.github/workflows/grading/check-all-results/README.md

```diff
@@ -0,0 +1,33 @@
+# Check All Results Action
+
+This action combines the results of multiple grading checks and returns a boolean indicating if all checks have passed.
+
+## Inputs
+
+The action expects input through environment variables. Each check result should be passed as an environment variable with a name starting with `check` (case insensitive).
+
+Each check value should be a JSON string in the following format:
+
+```json
+{
+  "name": "Display Name",
+  "passed": false,
+  "message": "Explanation of the result."
+}
+```
+
+## Example Usage
+
+```yml
+- name: Overall Check
+  id: check-combined-results
+  uses: skills/exercise-toolkit/.github/workflows/grading/check-all-results@cwb
```

Comment:
I'm not sure but I think this entire action can be replaced by

       ` if: steps.*.outcome != 'success'`